### DISCUSSION TITLE
Solax X1-Fit_G4

### DISCUSSION THREAD

**Discussion by anthonyc2020** [wills106/homeassistant-solax-modbus#579] [Q&A]
Hi I have taken the Plugin_solax.py and added FIT in most places that Hybride is located - Serial is F450xxxxx and got it mostly working - Could I ask that you add FIT to your source for Plugin_solax.py when convenient otherwise it doesn't recognise my inverter.

**Comment by cleveHEX** [comment]
You can send a pull request if you wish, but you have to guarantee that it works, not just randomly adding stuff without testing.

**Comment by wills106** [comment]
The FIT range is already supported:

https://github.com/wills106/homeassistant-solax-modbus/blob/86308cd44bc665497b603414b979659a97cdb343/custom_components/solax_modbus/plugin_solax.py#L5663-L5668

If you are missing any entities from the AC / FIT range you need to add in an extra AC and not FIT

For example:
https://github.com/wills106/homeassistant-solax-modbus/blob/86308cd44bc665497b603414b979659a97cdb343/custom_components/solax_modbus/plugin_solax.py#L3495-L3506

https://github.com/wills106/homeassistant-solax-modbus/blob/86308cd44bc665497b603414b979659a97cdb343/custom_components/solax_modbus/plugin_solax.py#L3505
Would become:
`allowedtypes = GEN3 | GEN4 | GEN | AC | HYBRID,`
